cache-config/t3c-apply/torequest/torequest.go

```diff
@@ -1030,6 +1034,45 @@ func (r *TrafficOpsReq) ProcessPackages() error {
 	return nil
 }
 
+func pkgMetaDataToMap(pmd []string) map[string]bool {
+	pkgMap := map[string]bool{}
+	for _, pkg := range pmd {
+		pkgMap[pkg] = true
+	}
+	return pkgMap
+}
+
+func pkgMatch(pkgMetaData []string, pk string) bool {
+	for _, pkg := range pkgMetaData {
+		if ok := strings.Contains(pk, pkg); ok {
+			return true
+		}
+	}
+	return false
+
+}
+
+func (r *TrafficOpsReq) ProcessPackagesWithMetaData(packageMetaData []string) error {
+	pkgs, err := getPackages(r.Cfg)
+	pkgMdataMap := pkgMetaDataToMap(packageMetaData)
+	if err != nil {
+		return errors.New("getting packages: " + err.Error())
+	}
+	for _, pkg := range pkgs {
+		fullPackage := pkg.Name + "-" + pkg.Version
+		if pkgMdataMap[fullPackage] {
+			log.Infof("package %v is assumed to be installed according to metadata file", fullPackage)
+			r.Pkgs[fullPackage] = true
+		} else if pkgMatch(packageMetaData, pkg.Name) {
+			log.Infof("package %v is assumed to be installed according to metadata, but doesn't match traffic ops pkg", fullPackage)
```

Comment:
nit: formatting parameter could be more specific; strings may use `%s` for better compile-time safety

Reply:
updaed to `%s`